pkg/volume/glusterfs/glusterfs_util.go

```diff
@@ -0,0 +1,58 @@
+package glusterfs
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+
+	"github.com/golang/glog"
+)
+
+// readGlusterLog will take the last 2 lines of the log file
+// on failure of gluster SetUp and return those so kubelet can
+// properly expose them
+// return nil on any failure
+func readGlusterLog(path string, podName string) error {
+
+	var line1 string
+	var line2 string
+	linecount := 0
+
+	glog.Infof("glusterfs: failure, now attempting to read the gluster log for pod", podName)
+
+	// Check and make sure path exists
+	if len(path) == 0 {
+		glog.Errorf("glusterfs: log file does not exist for pod: %s", podName)
+		return nil
+	}
+
+	// open the log file
+	file, err := os.Open(path)
+	if err != nil {
+		glog.Errorf("glusterfs: could not open log file for pod: %s", podName)
+		return nil
+	}
+	defer file.Close()
+
+	// read in and scan the file using scanner
+	// from stdlib
+	fscan := bufio.NewScanner(file)
+
+	// rather than guessing on bytes or using Seek
+	// going to scan entire file and take the last two lines
```

Comment:
not sure at this point, so far I've only been able to produce two line errors from gluster - there is a risk if the log file is shared or if even there are timestamped errors in the log file that if your latest error is only a single line, then you would get a previous error in your two lines...was thinking of doing some kind of quick match on the timestamps to make sure it's at least the same day  @rootfs